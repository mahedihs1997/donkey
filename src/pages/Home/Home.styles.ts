import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles({
  table: {
    minWidth: 650,
  },
  divider: {
    marginBottom: 40,
    marginTop: 40,
  },
});
