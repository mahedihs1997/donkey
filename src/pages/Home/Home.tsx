import {
  Box,
  Button,
  Card,
  Divider,
  List,
  ListItem,
  ListItemText,
  Typography,
} from '@material-ui/core';
import Cookie from 'js-cookie';
import React from 'react';

import { useStyles } from './Home.styles';

const Home = () => {
  const classes = useStyles();

  return (
    <div>
      <div className={classes.container}>
        <button
          type="button"
          onClick={() => {
            Cookie.set('test', 'Hi');
          }}
        >
          Set a damm cookie
        </button>
        <button
          type="button"
          onClick={() => {
            Cookie.remove('test');
          }}
        >
          Remove cookie
        </button>
        <Card className={classes.cardRoot}>
          <Typography variant="h5">3 Jobs Are unassigned!</Typography>
          <Box mt={2} />
          <Divider className={classes.divider} />
          <Box mt={2} />
          <List>
            <ListItem button>
              <ListItemText primary="ID: 7891234 Days: 7" />
            </ListItem>
            <ListItem button>
              <ListItemText primary="Spam" />
            </ListItem>
            <ListItem button>
              <ListItemText primary="Spam" />
            </ListItem>
          </List>
          <Box mt={2} />
          <Divider className={classes.divider} />
          <Box mt={2} />
          <Button variant="contained" color="primary">
            Assign Jobs
          </Button>
        </Card>
        <Card className={classes.cardRoot}>
          <Typography variant="h5">Monthly Reports</Typography>
          <Box mt={2} />
          <Divider className={classes.divider} />
          <Box mt={2} />
          <List>
            <ListItem button>
              <ListItemText primary="January MOT" />
            </ListItem>
            <ListItem button>
              <ListItemText primary="March Filer" />
            </ListItem>
            <ListItem button>
              <ListItemText primary="November Service" />
            </ListItem>
          </List>
          <Box mt={2} />
          <Divider className={classes.divider} />
          <Box mt={2} />
          <Button variant="contained" color="primary">
            Generate Report
          </Button>
        </Card>
      </div>
    </div>
  );
};

Home.getInitialProps = async () => {
  return {
    showLayout: true,
    title: 'Home',
  };
};

export default Home;
