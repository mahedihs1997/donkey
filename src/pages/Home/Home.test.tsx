import React from 'react';
import TestRenderer from 'react-test-renderer';

import Home from '.';

describe('<Home />', () => {
  it('should render without throwing an error and match snapshot', () => {
    const testRenderer = TestRenderer.create(<Home />);
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
