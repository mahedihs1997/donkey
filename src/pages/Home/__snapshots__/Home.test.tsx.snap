// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render without throwing an error and match snapshot 1`] = `
<div>
  <div
    className="makeStyles-container-3"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Set a damm cookie
    </button>
    <button
      onClick={[Function]}
      type="button"
    >
      Remove cookie
    </button>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-cardRoot-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5"
      >
        3 Jobs Are unassigned!
      </h5>
      <div
        className="MuiBox-root MuiBox-root-4"
      />
      <hr
        className="MuiDivider-root makeStyles-divider-2"
      />
      <div
        className="MuiBox-root MuiBox-root-5"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              ID: 7891234 Days: 7
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Spam
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Spam
            </span>
          </div>
        </div>
      </ul>
      <div
        className="MuiBox-root MuiBox-root-6"
      />
      <hr
        className="MuiDivider-root makeStyles-divider-2"
      />
      <div
        className="MuiBox-root MuiBox-root-7"
      />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Assign Jobs
        </span>
      </button>
    </div>
    <div
      className="MuiPaper-root MuiCard-root makeStyles-cardRoot-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5"
      >
        Monthly Reports
      </h5>
      <div
        className="MuiBox-root MuiBox-root-8"
      />
      <hr
        className="MuiDivider-root makeStyles-divider-2"
      />
      <div
        className="MuiBox-root MuiBox-root-9"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              January MOT
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              March Filer
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              November Service
            </span>
          </div>
        </div>
      </ul>
      <div
        className="MuiBox-root MuiBox-root-10"
      />
      <hr
        className="MuiDivider-root makeStyles-divider-2"
      />
      <div
        className="MuiBox-root MuiBox-root-11"
      />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Generate Report
        </span>
      </button>
    </div>
  </div>
</div>
`;
